src/components/MessageList/utils/getDateSeparators.ts

```diff
@@ -0,0 +1,75 @@
+import type { PaginatedMessageListContextValue } from '../../../contexts/paginatedMessageListContext/PaginatedMessageListContext';
+import type { ThreadContextValue } from '../../../contexts/threadContext/ThreadContext';
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../../types/types';
+
+export type GetDateSeparatorsParams<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = {
+  messages:
+    | PaginatedMessageListContextValue<At, Ch, Co, Ev, Me, Re, Us>['messages']
+    | ThreadContextValue<At, Ch, Co, Ev, Me, Re, Us>['threadMessages'];
+  hideDateSeparators?: boolean;
+};
+
+export type DateSeparators = {
+  [key: string]: Date;
+};
+
+export const getDateSeparators = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+>(
+  params: GetDateSeparatorsParams<At, Ch, Co, Ev, Me, Re, Us>,
+) => {
+  const { hideDateSeparators, messages } = params;
+  const dateSeparators: DateSeparators = {};
+
+  if (hideDateSeparators) {
+    return dateSeparators;
+  }
+
+  for (let i = 0; i < messages.length; i++) {
+    const previousMessage = messages[i - 1];
+    const message = messages[i];
+
+    if (
+      message.type === 'channel.event' ||
```

Comment:
`channel.event` this is some legacy stuff!! Not used anymore .. so can be removed.

Reply:
I am removing it from group styles too.

Reply:
Since you refactoring this, can you also take a look at `message.type !== 'message.read'` (in MessageList) . AFAIK, we don't have any such message type, nor we add this type in our SDK anywhere. We have event type as `message.read` but not message type. We can remove that as well I think